src/daft-core/src/array/ops/utf8.rs

```diff
@@ -60,6 +63,48 @@ fn is_valid_input_lengths(lengths: &[usize]) -> bool {
     }
 }
 
+fn check_valid_inputs<T, I>(
```

Comment:
`check_*` doesn't imply that it returns a result to me, so it was a little confusing to understand what the returned data was.

WDYT if instead it just returns `(all_null, size)` like so:

```
fn parse_inputs(...) -> DaftResult<(bool, usize)> {...}
```

Handling can then be left up to the caller, which has more information (e.g. dtype) to create the necessary arrays. Like this:

```
let (full_null, expected_size) = parse_inputs(...)?;
if full_null {
    return MyArrayType::full_null(dtype, size);
}

// You can also use the expected_size to do your broadcasting pretty easily
let self_iter = create_broadcasted_str_iter(self, expected_size);
let pattern_iter = create_broadcasted_str_iter(pattern, expected_size);

...

assert_eq!(expected_size, result_size)
```